src/sidebar/auth.js

```diff
@@ -1,122 +1,53 @@
-
-  var config = {
-    headers: headers,
-    // Skip JWT authorization for the token request itself.
-    skipAuthorization: true,
-  };
-
-  return $http.get(tokenUrl, config).then(function (response) {
-    return response.data;
-  });
-}
+var NULL_TOKEN = Promise.resolve(null);
 
 /**
  * Service for fetching and caching access tokens for the Hypothesis API.
  */
 // @ngInject
-function auth($http, flash, jwtHelper, session, settings) {
+function auth($http, jwtHelper, settings) {
 
-  var cachedToken = INITIAL_TOKEN;
+  var cachedToken = NULL_TOKEN;
 
   /**
-   * Fetches or returns a cached JWT API token for the current user.
+   * Fetch a new API token for the current logged-in user.
    *
-   * @return {Promise} - A promise for a JWT API token for the current
-   *                     user.
-   */
-  // @ngInject
-  function fetchOrReuseToken($http, jwtHelper, session, settings) {
-    function refreshToken() {
-      return fetchToken($http, session, settings).then(function (token) {
-        return token;
-      });
-    }
-
-    var userid;
-
-    return session.load()
-      .then(function (data) {
-        userid = data.userid;
-        if (userid === cachedToken.userid && cachedToken.token) {
-          return cachedToken.token;
-        } else {
-          cachedToken = {
-            userid: userid,
-            token: refreshToken(),
-          };
-          return cachedToken.token;
-        }
-      })
-      .then(function (token) {
-        if (jwtHelper.isTokenExpired(token)) {
-          cachedToken = {
-            userid: userid,
-            token: refreshToken(),
-          };
-          return cachedToken.token;
-        } else {
-          return token;
-        }
-      });
-  }
-
-  function clearCache() {
-    cachedToken = INITIAL_TOKEN;
-  }
-
-  /**
-   * Log out from the API and clear any cached tokens.
+   * The user is authenticated using their session cookie.
    *
-   * @return {Promise<void>} - A promise for when logout has completed.
+   * @return {Promise<string>} - A promise for a new JWT token.
    */
-  function logout() {
-    return session.logout({}).$promise
-      .then(function() {
-        clearCache();
-      })
-      .catch(function(err) {
-        flash.error('Log out failed!');
-        throw err;
-      });
+  function fetchToken() {
+    var tokenUrl = new URL('token', settings.apiUrl).href;
+    return $http.get(tokenUrl, {}).then(function (response) {
+      return response.data;
+    });
   }
 
   /**
-   * Return an access token for authenticating API requests.
+   * Fetch or return a cached JWT API token for the current user.
    *
-   * @return {Promise<string>}
+   * @return {Promise<string>} - A promise for a JWT API token for the current
+   *                             user.
    */
   function tokenGetter() {
-    return fetchOrReuseToken($http, jwtHelper, session, settings);
+    return cachedToken.then(function (token) {
```

Comment:
"cachedToken" leads you to expect it's a string or the actual token. Naming it to cachedTokenPromise or something that lends more toward what it is would make it better to understand when reading through